modules/doc/content/python/CSVDiff.md

```diff
@@ -0,0 +1,192 @@
+
+The 'GLOBAL VARIABLES' field allows you to change the tolerance for every field present in the CSV file. There are two key parameters; relative and floor. You can modify one or both of the
+values immediately following the parameter to suite your needs. You can also modify the tolerances for each individual field. In the case of our example, 'x' is the only field in our CSV
+file. To adjust only that field's tolerance values, we can add a parameter directly proceeding the 'x' label:
+
+```
+TIME STEPS relative 1 floor 0  # min: 0 @ t0  max: 0 @ t0
+
+GLOBAL VARIABLES relative 5.5e-06 floor 1e-11
+    x relative 5.5e-06   # min: 0.000e+00 @ t0          max: 0.000e+00 @ t0
+```
+
+The above change does nothing, as the relative error value we added is the same as the global relative error value. You can also add both relative and floor tolerances to this line. As well as comments and other logical statements:
+
+```
+GLOBAL VARIABLES relative 5.5e-06 floor 1e-11
+    # loosen tolerances for x
+    x floor 1e-8 relative 2e-04
+
+    # do not run differential tests for field y
+    !y
+
+    z
+```
+
+Here we added comments, loosened both the floor and error tolerances for field 'x'. Field 'y' will be ignored entirely. The 'z' field we left alone, and will end up using the global values set forth by the global variables header line.
+
+## A Real Example
+
+Consider the following two CSV files:
+
+File a:
+
+```
+x,y,z
+0,0,100
+1,0.000001,1
+```
+
+File b:
+
+```
+y,z,x
+0,100,0
+0.000001000005501,1,1
+```
+
+!alert note
+We purposely altered the field header to demonstrate CSVDiff's capability of correctly mapping the field labels between two files.
+
+If we run csvdiff.py on a and b, we see there is a small difference of 5.501e-06 for field 'y' at time step 1 (or simply put, row 1). Just a bit more than what our default global tolerance allows for:
+
+```
+> moose/scripts/csvdiff.py a b
+In file b: The values in column "y" don't match @ t1
+    relative diff:   0.000001 ~ 0.000001 = 0.000006 (5.501e-06)
+```
+
+We can create a comparison file to set forth new tolerances which will allow the two files to be considered identical. Start off by creating a comparison file using file 'a':
+
+```
+> moose/scripts/csvdiff.py --summary a > a.cmp
+```
+
+If we want to loosen the error tolerances for 'y', we would modify the comparison file like so:
+
+```
+TIME STEPS relative 1 floor 0  # min: 0 @ t0  max: 1 @ t1
+
+GLOBAL VARIABLES relative 5.5e-06 floor 1e-11
+    y relative 5.501e-06 # min: 0.000e+00 @ t0          max: 1.000e-06 @ t1
+    x                    # min: 0.000e+00 @ t0          max: 1.000e+00 @ t1
+    z                    # min: 1.000e+00 @ t1          max: 1.000e+02 @ t0
+```
+
+We can also instruct CSVDiff to ignore field 'y' all together:
+
+```
+TIME STEPS relative 1 floor 0  # min: 0 @ t0  max: 1 @ t1
+
+GLOBAL VARIABLES relative 5.5e-06 floor 1e-11
+    !y                   # min: 0.000e+00 @ t0          max: 1.000e-06 @ t1
```

Comment:
Wow - is this something that's supported by Exodiff?

Reply:
Hmmm I am not sure. I'll test it. If it does not work, I suppose I should remove this feature from csvdiff?

Or, just update our statement about both summery files being interchangeable...

Reply:
What did you find out about this syntax? Does this exist in Exodiff? @bwspenc - do you know?

Reply:
I think it does work. Here is my custom comparison file:
```
[#]> cat simple.cmp
COORDINATES absolute 1.e-6    # min separation not calculated
TIME STEPS relative 1.e-6 floor 0.0     # min:               0 @ t1 max:               1 @ t2
# No GLOBAL VARIABLES
NODAL VARIABLES relative 1.e-6 floor 0.0
	!u  # min:               0 @ t1,n1	max:               1 @ t2,n21
# No ELEMENT VARIABLES
# No NODESET VARIABLES
# No SIDESET VARIABLES
```

And the results when I ask exodiff to use it:
```
[#]> exodiff -f  simple.cmp gold/simple_diffusion_out.e simple_diffusion_out.e 
<... trimmed ...>
Nodal coordinates will be compared .. tol:    1e-06 (absolute), floor:        0
Time step values will be compared .. tol:    1e-06 (relative), floor:        0
No Global variables on either file.
Nodal variables will not be compared. <--- THIS IS STATED
No Element variables on either file.
No Element Attribute variables on either file.
No Nodeset variables on either file.
No Sideset variables on either file.
<... trimmed ...>

exodiff: Files are the same
```

As opposed to this when I don't:
```
<... trimmed ...>
Nodal coordinates will be compared .. tol:    1e-06 (absolute), floor:        0
Time step values will be compared .. tol:    1e-06 (relative), floor:        0
No Global variables on either file.
Nodal variables to be compared:
	u                                tol:    1e-06 (relative), floor:        0
No Element variables on either file.
No Element Attribute variables on either file.
No Nodeset variables on either file.
No Sideset variables on either file.
Sideset Distribution Factors will be compared .. tol:    1e-06 (relative), floor:        0
<... trimmed ...>
exodiff: Files are the same
```

Reply:
OK!